src/battle/overlay_12_0224E4FC.c

```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+            ctx->statChangeType = 3;
+            ctx->battlerIdStatChange = ctx->battlerIdAttacker;
+            ctx->battlerIdWork = ctx->battlerIdTarget;
+            *script = 106;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_AFTERMATH:
+        if (ctx->battlerIdTarget == ctx->battlerIdFainted &&
+            GetBattlerAbility(ctx, ctx->battlerIdAttacker) != ABILITY_MAGIC_GUARD &&
+            !CheckAbilityActive(bsys, ctx, CHECK_ABILITY_ALL_HP, 0, ABILITY_DAMP) && 
+            !(ctx->linkStatus2 & 0x10) &&
+            ctx->battleMons[ctx->battlerIdAttacker].hp &&
+            !(ctx->moveStatusFlag & MOVE_STATUS_FAIL) &&
+            (ctx->unk_334.moveData[ctx->moveNoCur].unkB & 1)) {
+            ctx->hpCalcWork = DamageDivide(ctx->battleMons[ctx->battlerIdAttacker].maxHp * -1, 4);
+            ctx->battlerIdWork = ctx->battlerIdAttacker;
+            *script = 193;
+            ret = TRUE;
+        }
+        break;
+    default:
+        break;
+    }
+    return ret;
+}
+
+BOOL CheckStatusHealAbility(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId, int flag) {
+    BOOL ret = FALSE;
+    
+    switch (GetBattlerAbility(ctx, battlerId)) {
+    case ABILITY_IMMUNITY:
+        if (ctx->battleMons[battlerId].status & STATUS_POISON_ALL) {
+            ctx->msgWork = 1;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_OWN_TEMPO:
+        if (ctx->battleMons[battlerId].status2 & STATUS2_CONFUSION) {
+            ctx->msgWork = 5;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_LIMBER:
+        if (ctx->battleMons[battlerId].status & STATUS_PARALYSIS) {
+            ctx->msgWork = 3;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_INSOMNIA:
+    case ABILITY_VITAL_SPIRIT:
+        if (ctx->battleMons[battlerId].status & STATUS_SLEEP) {
+            ctx->msgWork = 0;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_WATER_VEIL:
+        if (ctx->battleMons[battlerId].status & STATUS_BURN) {
+            ctx->msgWork = 2;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_MAGMA_ARMOR:
+        if (ctx->battleMons[battlerId].status & STATUS_FREEZE) {
+            ctx->msgWork = 4;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_OBLIVIOUS:
+        if (ctx->battleMons[battlerId].status2 & STATUS2_ATTRACT_ALL) {
+            ctx->msgWork = 6;
+            ret = TRUE;
+        }
+        break;
+    case ABILITY_UNBURDEN:
+        //resets the item flag in the case where the Pokemon acquires an item via Thief or Trick
+        if (ctx->battleMons[battlerId].item) {
+            ctx->battleMons[battlerId].unk88.knockOffFlag = TRUE;
+        }
+        break;
+    }
+    if (ret == TRUE) {
+        ctx->battlerIdWork = battlerId;
+        ctx->abilityWork = GetBattlerAbility(ctx, battlerId);
+        if (!flag) {
+            ReadBattleScriptFromNarc(ctx, NARC_a_0_0_1, 221);
```

Comment:
no naix for the narc?

Reply:
Nope, we haven't decompiled the battle script narc yet for whatever reason